zip-0308.rst

```diff
@@ -0,0 +1,365 @@
+transactions from different users, contrary to the supposed adversary's
+inference).
+
+Making the chosen amount inclusive of the fee avoids leaving any unmigrated
+funds at the end, in the case where the original amount to migrate was a
+multiple of 0.01 ZEC.
+
+
+Other design decisions
+----------------------
+
+We assume use of the normal wallet note selection algorithm and change
+handling. Change is sent back to the default address, which is the z-address
+of the first selected Sprout note. The number of JoinSplits will therefore be
+the same as for a normal transaction sending the same amount with the same
+wallet state. Only the ``vpub_out`` of the last JoinSplit will be nonzero.
+There will always be exactly one Sapling Output.
+
+The expiration height is chosen as for other transactions.
+
+The fee for each migration transaction MUST be 0.0001 ZEC. This fee is taken
+from the funds to be migrated.
+
+Some wallets by default add a "developer fee" to each transaction, directed
+to the developer(s) of the wallet. This is typically implemented by adding
+the developer address as an explicit output, so if migration transactions are
+generated internally by ``zcashd``, they will not include the developer fee.
+We strongly recommend *not* patching the ``zcashd`` code to add the developer
+fee output to migration transactions, because doing so partitions the anonymity
+set between users of that wallet and other users.
+
+There MUST NOT be any transparent inputs or outputs, or Sapling Spends, in a
+migration transaction.
+
+The ``lock_time`` field MUST be set to 0 (unused).
+
+
+Open questions
+--------------
+
+The above strategy has several "magic number" parameters:
+
+* the interval between batches (500 blocks)
+* the maximum number of transactions in a batch (5)
+* the distribution of exponents (uniform integer in 6..9)
+* the distribution of mantissae (uniform integer in 1..99).
+
+These have been chosen by guesswork. Should we change any of them?
+
+In particular, if the amount to migrate is large, then this strategy can
+result in large amounts (up to 990 ZEC, worth USD ~49500 at time of writing)
+transferred in each transaction. This leaks the fact that the transaction was
+sent by a user who has at least that amount.
+
+The strategy does not migrate any remaining fractional amount less than
+0.01 ZEC (worth USD ~0.50 at time of writing). Is this reasonable?
+
+In deciding the amount to send in each transaction, the strategy does not
+take  account of the values of individual Sprout notes, only the total amount
+remaining to migrate. Can a strategy that is sensitive to individual note
+values improve privacy?
+
+An adversary may attempt to interfere with the view of the block chain seen
+by a subset of nodes that are performing migrations, in order to cause those
+nodes to send migration batches at a different time, so that they may be
+distinguished. Is there anything further we can do to mitigate this
+vulnerability?
```

Comment:
Assuming we aren't concerned here about a 50%+1 adversary, the problematic attacks here involve network partitioning, BGP route manipulation, etc. in order to delay block propagation. This would exacerbate an existing problem, which is the inherent propagation time across the network. We do not assume network privacy, so I'm not sure there's much we can do here. 

If we, for example, started broadcasting transactions at some randomised interval after observing a boundary block, that would obscure the exact time we detected the block (to within some delta), with the downside that there is less time for our transactions to propagate and potentially get mined into the target block. I'm not sure the delta can be large enough to thwart a motivated network adversary, without compromsing the ability to get transactions mined promptly. But I don't have any hard numbers for this, and maybe it is possible. It seems more complicated to implement, regardless.

Reply:
I agree there's not much we can do here.